include/wx/cpp.h

```diff
@@ -114,6 +114,26 @@
     #define __WXFUNCTION__ __func__
 #endif /* __WXFUNCTION__ already defined */
 
+/*
+    Display the current function's full signature, if available.
+
+    Falls back to the function name (i.e., __func__) if not available.
+ */
+#ifndef __WXFUNCTION_SIG__
+    #if __cplusplus >= 202002L
+        #include <source_location>
+        // actually displays the signature, not just the name
+        #define __WXFUNCTION_SIG__ std::source_location::current().function_name()
+    #elif defined(__VISUALC__)
+        #define __WXFUNCTION_SIG__ __FUNCSIG__
+    #elif defined(__clang__)
+        #define __WXFUNCTION_SIG__ __func__
```

Comment:
I thought clang supported `__PRETTY_FUNCTION__` too, so do we really need to have a separate test for it here?

Reply:
Indeed it does. I read people complaining that it is missing that, but I verified that it does on compiler explorer. Fixed.